### ISSUE TITLE
Frequent pauses, loss of sound, when playing a already downloaded torrent through WebTorrent-VLC.

### ISSUE THREAD

**Issue by ethaniel** [feross/webtorrent-desktop#660]
**What version of WebTorrent Desktop?** (See the 'About WebTorrent' menu)
0.7.2

**What operating system and version?**
Mac El Capitan

**What did you do?**
I fully downloaded a torrent with 10 videos through WTD. Clicked "play" on a random video. WTD told me that audiocodec is unsupported and suggested to play in VLC. I clicked "Play in VLC". 

**What did you expect to happen?**
I expected a smooth playback, since the torrent was already downloaded.

**What actually happened?**
Frequent stops (video and audio freezes for a few seconds). Then the playing resumed with artifacts which went away after a few seconds. The stops became more frequent closer to the end of the video.

**Additional info**
I've noticed that VLC was playing the video through the socket http://localhost:56535/2.
When I opened the video file from the download folder into VLC directly (without WTD), the playback was smooth.

I suspect that the error is somewhere inside WTD transmitting the video stream from the file into the socket.

**Comment by dcposch** [comment]
> I've noticed that VLC was playing the video through the socket http://localhost:56535/2.
When I opened the video file from the download folder into VLC directly (without WTD), the playback was smooth.


Very strange. WebTorrent Desktop just starts VLC with a command-line argument pointing to that same local HTTP server, so those two things *should* be exactly the same.

Thanks for the bug report! More reliable video playback is our top priority right now.

Goals:
* Reliable playback in VLC
* No spurious codec errors. Currently the mechanism by which WTD detects if a video uses an unsupported codec sometimes has false positives.
* Option to play videos via Airplay/Chromecast in cases where the cast device supports the codec, but the <video> tag doesn't. (Currently, your only option is "Play in VLC")

And, ideally:
* More codec support, so we don't have to show "Play in VLC" as often (or ever)

**Comment by ethaniel** [comment]
> Very strange. WebTorrent Desktop just starts VLC with a command-line argument pointing to that same local HTTP server, so those two things should be exactly the same.

Just to make it clear:

1. If the video is being played (through http://) while still downloading from torrent - everything is OK and smooth.
2. If the video is being played (through http://) after finished downloading from torrent - sometimes choppy. 
3. If the video is being played directly from a file where it was downloaded into - when I open it from Finder for example - everything is OK and smooth.

**Comment by feross** [comment]
I think I've seen this before, but it's not clear what caused it, and I've been unable to track it down.

**Comment by feross** [comment]
@ethaniel Have you had any issues with this in the most recent version of WebTorrent Desktop (0.9.0)? I haven't seen this issue in a while, so it might be gone now.

**Comment by ethaniel** [comment]
I am still having same issues in 0.10.0 (0.96.0).

**Comment by feross** [comment]
Okay, re-opening.

**Comment by ethaniel** [comment]
We now have a new issue.
I am watching the torrent that is being downloaded. I was directed to use the VLC player. But now the VLC player now randomly stops and closes while I'm watching.
0.10.0 (0.96.0)

**Comment by feross** [comment]
VLC's http streaming is extremely flaky. Seeking to certain time codes in the stream RELIABLY makes VLC crash.

I just spent the last hour trying to figure out if there's any way for us to make it more reliable, but it appears that for some files VLC just makes a guess about where to seek to (translating the time code to a byte offset into the file) and when it guesses wrong, it just aborts.

I confirmed that this is not just an issue with WebTorrent's http server. I started an http server with [`ecstatic`](https://www.npmjs.com/package/ecstatic) and it had the same issue playing back `sintel.mp4`.

There's also [this issue](https://bugs.launchpad.net/ubuntu/+source/vlc/+bug/790216) where a VLC maintainer says that they don't intend to make VLC retry when there are http streaming errors. 

We're shipping the ability to specify other external video players like `mplayer` and `mpv` in the next version of WebTorrent Desktop. I recommend giving one of these others a shot. They appear to work more reliably.

Long term solution: make playing back everything in WebTorrent Desktop work more reliably.
